scale_build/packages/build.py

```diff
@@ -71,7 +70,7 @@ def _build_impl(self):
                 predep_cmd = predep_entry['command']
                 skip_cmd = False
                 for env_var in predep_entry['env_checks']:
-                    if os.environ.get(env_var['key']) != env_var['value']:
+                    if all(f(env_var['key']) != env_var['value'] for f in (os.environ.get, self.env.get)):
```

Comment:
You forgot `APT_ENV` here.
Can we make a single method that will construct resulting `env` for command and check against it instead?